### PR TITLE
go/deps: Fix bad update

### LINKED ISSUE #31608: go network example broken by golang update
Seems a dependabot update this morning has broken the golang-network example

ill take a look at fixing forward, otherwise revert the update

### PR DESCRIPTION
**Pull request by phlax**
Fix #31608 

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @doujiang24

**Comment by phlax** [comment]
@adisuissa this is the product of running `go mod tidy` on the root and example go files - not 100% about the fix - but it works locally testing the currently broken example 

not sure if anything else needs this and confused that dependabot has created this issue

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
Let's wait for @doujiang24 to see if they have some comments

**Comment by phlax** [comment]
@adisuissa this required an additional update

**Review by adisuissa — APPROVED** [review]
LGTM

**Comment by phlax** [comment]
on the basis that it appears to resolve the issue that all prs are currently hitting - im gonna land - fwiw, i think the issue may have been related to bumping node 21.4 -> 21.5 in containers

File: go.mod [hunk 1]
```diff
@@ -2,4 +2,23 @@ module github.com/envoyproxy/envoy
...
+	github.com/census-instrumentation/opencensus-proto v0.4.1
+	github.com/cncf/xds/go v0.0.0-20231128003011-0fa0005c9caa
+	github.com/envoyproxy/go-control-plane v0.12.0
+	github.com/envoyproxy/protoc-gen-validate v1.0.2
+	github.com/golang/protobuf v1.5.3
+	github.com/prometheus/client_model v0.5.0
+	go.opentelemetry.io/proto/otlp v1.0.0
+	google.golang.org/genproto/googleapis/api v0.0.0-20240102182953-50ed04b92917
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20240102182953-50ed04b92917
+	google.golang.org/grpc v1.60.1
```
Comment (doujiang24) [inline go.mod:15]: Hello @phlax , these requires looks a bit weird. this is what I got, after `go mod tidy`:
```
module github.com/envoyproxy/envoy

go 1.20

require google.golang.org/protobuf v1.32.0

require github.com/google/go-cmp v0.5.9 // indirect
```

are there some untracked go source files in your side? i.e. some go files for local testing under the repo dir.
`go mod tidy` will generate requires for them too.

Comment (phlax) [reply go.mod:15]: unlikely

ftr this is what i ran

```console
docker run --rm -it -v $PWD:/workspace --workdir /workspace golang:1.20 go mod tidy
docker run --rm -it -v $PWD:/workspace --workdir /workspace/examples/golang-network/simple golang:1.20 go mod tidy
docker run --rm -it -v $PWD:/workspace --workdir /workspace/examples/golang-http/simple golang:1.20 go mod tidy
```

Comment (phlax) [reply go.mod:15]: if the deps generated are not correct - then lets update - as long as it actually works

Comment (doujiang24) [reply go.mod:15]: okay, created a PR: https://github.com/envoyproxy/envoy/pull/31640
it's regenerated by: `docker run --rm -it -v $PWD:/workspace --workdir /workspace golang:1.20 go mod tidy`

**Pull request merged**